### PR TITLE
cppstd setting at first level

### LINKED ISSUE #1799: Consider setting for C++ std
Compiling the same source code, with the same compiler and version, activating or deactivating C++ std features might lead to different native code (type sizes, etc).

### PR DESCRIPTION
**Pull request by lasote**
- [x] Refer to the issue that supports this Pull Request. Closes #1799 
- [x] I've read the [Contributing guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [x] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. Also adding a description of the changes in the ``changelog.rst`` file. https://github.com/conan-io/docs

https://github.com/conan-io/docs/pull/548

### CODE REVIEW

File: conans/client/build/cppstd_flags.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+               "Visual Studio": _visual_cppstd_default(compiler_version)}.get(str(compiler), None)
+    return default
+
+
+def _gcc_cppstd_default(compiler_version):
+    if Version(compiler_version) < "6.1":
+        return "98gnu"
+    if Version(compiler_version) >= "6.1":
+        return "14gnu"
+    return None
```
Comment (SSE4) [inline conans/client/build/cppstd_flags.py]: there are only two cases, right? looks excessive
better write something like `return '98gnu' if Version(compiler_version) else '14gnu'`

File: conans/client/build/cppstd_flags.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+def _cppstd_clang(clang_version, cppstd):
+    """
+    Inspired in:
+    https://github.com/Kitware/CMake/blob/
+    1fe2dc5ef2a1f262b125a2ba6a85f624ce150dd2/Modules/Compiler/Clang-CXX.cmake
+    """
+    v98 = v98gnu = v11 = v11gnu = v14 = v14gnu = v17 = v17gnu = None
+
+    if not Version(clang_version) < "2.1":
```
Comment (SSE4) [inline conans/client/build/cppstd_flags.py]: may be reversed condition is easier to read?
`if Version(clang_version) >= "2.1":`

Comment (lasote) [reply conans/client/build/cppstd_flags.py]: Yes, I followed too much the CMake code, that as usual, is creepy.

File: conans/client/build/cppstd_flags.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
+from conans.model.version import Version
+
+
+def available_cppstd_versions(compiler, compiler_version):
+    ret = []
+    stds = ["98", "98gnu", "11", "11gnu", "14",  "14gnu", "17", "17gnu"]
```
[the remark this answers is not in the document]
Comment (lasote) [reply conans/client/build/cppstd_flags.py]: Ok, I thought it is more readable but yes, better gnu11.

**Comment by lasote** [comment]
Fixed test (hopefully) and added docs. About the "experimental" ok, you little coward hahhaha

**Pull request merged**